proto/proto_conn.c

```diff
@@ -390,11 +390,11 @@ callback_pipestatus(void * cookie)
 
 	/* If we have an error in either direction, kill the connection. */
 	if ((C->stat_f == -1) || (C->stat_r == -1))
-		return (proto_conn_drop(C));
+		return (proto_conn_drop(C, PROTO_CONN_ERROR));
 
 	/* If both directions have been shut down, kill the connection. */
 	if ((C->stat_f == 0) && (C->stat_r == 0))
-		return (proto_conn_drop(C));
+		return (proto_conn_drop(C, PROTO_CONN_ERROR));
```

Comment:
... which would be used here, since cleaning up after both ends of a connection have been shut down is absolutely not an error.